src/index.js

```diff
@@ -316,20 +337,25 @@ class JanusAdapter {
     debug("pub waiting for sfu");
     await handle.attach("janus.plugin.sfu");
 
+    this.associate(conn, handle);
+
+    debug("pub waiting for data channels & webrtcup");
+    var webrtcup = new Promise(resolve => handle.on("webrtcup", resolve));
```

Comment:
Technically we should also handle the error case while waiting for "webrtcup".

Reply:
not going to change these semantics for now